sync2jira/downstream_issue.py

```diff
@@ -152,6 +152,68 @@ def execute_snowflake_query(issue):
     return results
 
 
+def _get_field_id_by_name(client, field_name):
+    """
+    Convert a human-readable custom field name to its JIRA field ID.
+
+    :param jira.client.JIRA client: JIRA client
+    :param str field_name: Human-readable field name (e.g., "Story Points", "Epic Link")
+    :returns: Field ID (e.g., "customfield_12310243") or None if not found
+    :rtype: Optional[str]
+    """
+    # Cache field mappings per client to avoid repeated API calls
+    cache_key = id(client)
+    if cache_key not in jira_cache:
+        jira_cache[cache_key] = {}
+
+    # Check cache first
+    if field_name in jira_cache[cache_key]:
+        return jira_cache[cache_key][field_name]
+
+    # Fetch all fields from JIRA
+    try:
+        all_fields = client.fields()
+        name_map = {field["name"]: field["id"] for field in all_fields}
+
+        # Cache the entire mapping
+        jira_cache[cache_key] = name_map
+
+        # Return the field ID if found
+        field_id = name_map.get(field_name)
+        if field_id:
+            log.debug(f"Resolved field name '{field_name}' to ID '{field_id}'")
+        else:
+            log.warning(f"Field name '{field_name}' not found in JIRA fields")
+        return field_id
+    except Exception as e:
+        log.error(f"Error fetching JIRA fields: {e}")
+        return None
+
+
+def _resolve_field_identifier(client, field_identifier):
+    """
+    Resolve a field identifier (either a name or an ID) to a field ID.
+    If the identifier is already an ID (starts with 'customfield_'), return it as-is.
+    Otherwise, treat it as a name and convert it to an ID.
+
+    :param jira.client.JIRA client: JIRA client
+    :param str field_identifier: Field name (e.g., "Story Points") or ID (e.g., "customfield_12310243")
+    :returns: Field ID or None if not found
+    :rtype: Optional[str]
+    """
+    # If it's already an ID (starts with 'customfield_' or is a standard field), return as-is
+    if field_identifier.startswith("customfield_") or field_identifier in [
+        "priority",
+        "assignee",
+        "summary",
+        "description",
+    ]:
```

Comment:
A stylistic nit:  prefer immutable objects to mutable ones, especially for values which are constant -- use a `tuple` instead of a `list` here:
```suggestion
    if field_identifier.startswith("customfield_") or field_identifier in (
        "priority",
        "assignee",
        "summary",
        "description",
    ):
```